### ISSUE TITLE
"terraform console" incorrectly omits object and map elements whose values are null

### ISSUE THREAD

**Issue by moio** [hashicorp/terraform#23451]
### Terraform Version
Terraform v0.12.16

### Steps to Reproduce
```
$ terraform console
> {a=null}
{}
> keys({a=null})
[
  "a",
]
> keys({})
[]
```

### Expected Behavior

Either `{a=null} != {}`, or `keys({a=null}) == []`.

**Comment by moio** [comment]
Somewhat also suprisingly (maybe I should open a separate issue?)

```
$ terraform console
> {a=null}["a"]
null
> {}["a"]
Error: Invalid index
```

**Comment by apparentlymart** [comment]
I have a feeling that the bug here is in the object/map rendering code in `terraform console`: null values _are_ allowed in objects and maps, and are distinct from the attribute/element being absent, because Terraform's type system is a superset of JSON which makes a similar distinction.

In other words, the first step in the reproduction is the incorrect one. Should be:

```
> {a=null}
{
  a = null
}
```

It looks like this is just a cosmetic bug in the `terraform console` UI rather than an inconsistency in the type system itself, but still definitely worth a fix because indeed it's pretty confusing and misleading.

The reason for this inconsistency might be because the `terraform console` printing logic is similar to the `terraform plan` diff rendering. In the case of resource arguments `null` _is_ the same as unset, so this behavior makes sense in that context but not in the `terraform console` context where we're working with normal values.

**Comment by RulerOf** [comment]
> It looks like this is just a cosmetic bug in the `terraform console` UI rather than an inconsistency in the type system itself, but still definitely worth a fix because indeed it's pretty confusing and misleading.

_Definitely_ worth fixing. I threw out my entire plan for handling certain module inputs that I was working on when I filed #22316 because I thought that `merge()` was just broken due to how the console behaved.

I _can_ see now that it was clearly a console issue though:

```
> merge({"a"=1,"b"=2},{"a"=null,"c"=3}) 
{
  "b" = 2
  "c" = 3
}
> keys(merge({"a"=1,"b"=2},{"a"=null,"c"=3}))
[
  "a",
  "b",
  "c",
]
```

**Comment by alisdair** [comment]
This is fixed in #26189, which will be part of the Terraform 0.14.0 release.

Sample session with a build from today:

```text
$ terraform console
> {a=null}
{
  "a" = null
}
> merge({"a"=1,"b"=2},{"a"=null,"c"=3})
{
  "a" = null
  "b" = 2
  "c" = 3
}
```

**Comment by moio** [comment]
:tada: 

Thanks!!!
